libbeat/outputs/redis/redis.go

```diff
@@ -252,17 +252,36 @@ func (out *redisOutput) UpdateLocalTopologyMap(conn redis.Conn) {
 
 func (out *redisOutput) PublishEvent(
 	signal outputs.Signaler,
-	ts time.Time,
+	opts outputs.Options,
 	event common.MapStr,
 ) error {
-	return out.BulkPublish(signal, ts, []common.MapStr{event})
+	return out.BulkPublish(signal, opts, []common.MapStr{event})
 }
 
 func (out *redisOutput) BulkPublish(
 	signal outputs.Signaler,
-	ts time.Time,
+	opts outputs.Options,
 	events []common.MapStr,
 ) error {
+	if !opts.Guaranteed {
+		err := out.doBulkPublish(events)
+		outputs.Signal(signal, err)
+		return err
+	}
+
+	for {
+		err := out.doBulkPublish(events)
+		if err == nil {
+			outputs.SignalCompleted(signal)
+			return nil
+		}
+
+		// TODO: add backoff
+		time.Sleep(1)
```

Comment:
No plans to add anything to redis module right now. The redis module has become orphaned and basically needs to be rewritten to use the outputs/modes module like elasticsearch and logstash outputs do.